### ISSUE TITLE
After loading game Scene don't return to state form save

### ISSUE THREAD

**Issue by Jeremi360** [rakugoteam/Rakugo-archive#85]
Solutions is to use PackedScene with ResourceSaver and ResourceLoader
here is example of this: 
https://godotengine.org/qa/31138/saving-a-dictionary-of-scene-states

**Comment by Jeremi360** [comment]
Today I got an idea - there is simpler and easier way to fix it- just `inst2dict()` to save state of nodes that was add do ren's global variables by `node_link()` (now there is only save path to node) and then use `get_property_list()` and `set()` to set it after load.
What you think @ivandardi?

**Comment by jeremi360** [comment]
I just figure that is simple to fix just we need just add a timer which will wait about 0.5s after hide in game menu to unpause the game

**Comment by jebedaia360** [comment]
I see that use PackedScene and ResourceSaver will be better and simpler.
We should do it by have one variable with root node of current scene and list of nodes of other scenes to save

**Comment by Jeremi360** [comment]
I need to use `PackedScene` with `ResourceSaver `and `ResourceLoader` as dict - unfortunately `inst2dict() `only give vars define in script

**Comment by Jeremi360** [comment]
It is not a bug, it properly save/load data, but it just don't save scene state - it will be hard to do and have many disadvantages.
Just use build rakugo_vars mechanics to save what you need I also added on "loaded" signal to help with it.

**Comment by Jeremi360** [comment]
Yes this solution is good enough.
Here are examples:
- [examples/Test/TestAnimPlayer.gd](https://github.com/jebedaia360/Rakugo/blob/master/examples/Test/TestAnimPlayer.gd)
- [examples/Test/TestNode.gd](https://github.com/jebedaia360/Rakugo/blob/master/examples/Test/TestNode.gd)
- [examples/Test/TestNode2D.gd](https://github.com/jebedaia360/Rakugo/blob/master/examples/Test/TestNode2D.gd)

**Comment by Jeremi360** [comment]
This solution for some reason stop to work - save don't save all values of variables for some reason :(

**Comment by Jeremi360** [comment]
To replicate the bug:
1. run project
2. choose "new game"
3. go forward until get to part with testing animation
4. save game
5. close game
6. run project
7. load saved game
8. see if animation is playing
9. see if you can go further

**Comment by MatiasVME** [comment]
Hello, the error is with the persistence node ... when loading the data is erased. I did a test in beta11.

....

Well, I fixed that, but it only happened when you didn't use profiles.

I will try to do several tests for the persistence node to see if everything is working correctly.

**Comment by Jeremi360** [comment]
@MatiasVME Thanks for your help.
How you discover that "when loading the data is erased" ?
I don't know why bug is happening - I didn't change anything in side save/load methods for months.

**Comment by Jeremi360** [comment]
Ups!

**Comment by MatiasVME** [comment]
Let me explain... My english is bad..

I found a bug in PersistenceNode, but this only affects projects that don't use this:

Persistence.get_data("blabla") # No error

They're using this:

$Persistence.get_data() # Error

The PersistenceNode in the Rakugo project should not have problems. (I think)

Yesterday I was trying to see the error and I don't know what's going on.

**Comment by Jeremi360** [comment]
@MatiasVME I use `$Persistence.get_data()`, so I have an error
and I should make PersistenceNode as singleton yes?
Also I see that you did update then addon should I update it?

**Comment by MatiasVME** [comment]
Just change the following line:

https://github.com/jebedaia360/Rakugo/blob/9e62ec328790f654a2351a9334016dfedf737cde/addons/PersistenceNode/persistence_node.gd#L410

To:

```
if profile_name == "":
```

But I don't think that solves the rakugo error.

In a while I'm going to do some tests with PersistenceNode to see if everything is working correctly.

Edit:
And singleton is for the range, there is no other effect with it.

**Comment by MatiasVME** [comment]
Also so I saw you're not using get_data() without arguments.

```
matias@matias-Lenovo-ideapad-110-14ISK:/media/disco/git/Ren$ grep -rni 'get_data()'
gui/Screens/Screens.gd:159:	return get_viewport().get_texture().get_data()
```

**Comment by Jeremi360** [comment]
@MatiasVME I will fix this and update to 1.4.0 version of  PersistenceNode.
Next time when you update PersistenceNode tell me.

**Comment by MatiasVME** [comment]
This is the last version and the tests are correct.

https://github.com/MatiasVME/Persistence/releases/tag/v1.4.0-Godot.3.1

**Comment by Jeremi360** [comment]
@MatiasVME "Also so I saw you're not using get_data() without arguments" ?
`gui/Screens/Screens.gd:159:
return get_viewport().get_texture().get_data()` - This is correct code it is not connected with yours PersistenceNode.

**Comment by Jeremi360** [comment]
Bug continues to occur - it seems that for some reason not, all vars are saved.
I will try to make some code in `main.gd` less clutter - maybe it will fix it or make it easy to fix.

**Comment by MatiasVME** [comment]
That's all I was saying, since the grep command shows all the strings in which 'get_data()' is used in the project. 

The line:
gui/Screens/Screens.gd:159: return get_viewport().get_texture().get_data()

It is not related to persistence node.

If you run it with beta7 or 8, does it work?

**Comment by Jeremi360** [comment]
for me this line worked in every beta, also it is impossible to use this 'get_data()` with any arguments.

**Comment by Jeremi360** [comment]
I update to PersistenceNode 1.4.0 , and made code of rakugo simpler, cleaner and less clutter.
Switched PersistenceNode to Text mode to check if it fix any thing with save/load.
And now when I try to load I get this: 
```
[PersistenceNode] El archivo no existe: user://saves/default.save  
[PersistenceNode] retorna false.  
[PersistenceNode] El archivo no existe: user://saves/default.save  
[PersistenceNode] load_profile_default retorna false.  
```

**Comment by MatiasVME** [comment]
I've been watching, and I don't know what's gonna happen. S:

**Comment by Jeremi360** [comment]
I fixed PersistenceNode profile name checking, and it save,
 but for some reason characters after load are null,
 and some variables was saved with starting value - but it all is saved in the same way :(
I will try write better RakugoVar types - maybe it will fix some thing.

**Comment by Jeremi360** [comment]
I try check some older version of rakugo, but all of theme has this bug :/
It seams that it was only working because of some bug in godot, which was fixed in 3.1

**Comment by MatiasVME** [comment]
But do you think it's the PersistenceNode?

The tests are fine, if you think a test is missing you can add it to the repo de persistence node.

**Comment by Jeremi360** [comment]
Is there test for dictionary and array?
I save big dictionary with alot of dictionaries in it - maybe I should do it easier - saves these dictionaries with as data with out this one extra dictionary that's keep them together

**Comment by MatiasVME** [comment]
I've previously tested with giant dictionaries (in my game Spaceship Shooter RPG Deluxe). But I haven't tested the PeristenceNode version for 3.1 with giant dictionaries.

I'll try to see what happens in Rakugo later.

**Comment by Jeremi360** [comment]
I will compare current *main.gd* with *main.gd* from 1.0 - maybe this way I will find what is wrong.

**Comment by Jeremi360** [comment]
It didn't help.
But I got idea to add saved nodes to "save" group and use it to update its var base before save.
I also need to check all ifs in jump()

**Comment by Jeremi360** [comment]
Fixed `jump()` now after load you can go further :D #131

**Comment by Jeremi360** [comment]
It fixed in `resource-save` branch :D ,
 but bug with that you can't go further is back
https://youtu.be/8amzuitYWtI

**Comment by Jeremi360** [comment]
> but bug with that you can't go further is back
 
:arrow_up:  fixed :D

But this save system/load need few tweaks

**Comment by Jeremi360** [comment]
Fixed an merged #135
